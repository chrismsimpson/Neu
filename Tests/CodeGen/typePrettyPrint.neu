/// Expect:
/// - output: "Value.bar(values: ["hello"])\nRect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))\nFoo.baz(100)\n"

class Size {
    width: Int64
    height: Int64
}

class Point {
    x: Int64
    y: Int64
}

class Rect {
    location: Point
    size: Size
}

enum Value {
    foo
    bar(values: [String])
}

enum Foo {
    bar
    baz(Int64)
}

func main() {
    let e = Value.bar(values: ["hello"])
    printLine("{}", e)

    let r = Rect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))
    printLine("{}", r)

    let foo = Foo.baz(100)
    printLine("{}", foo)
}
