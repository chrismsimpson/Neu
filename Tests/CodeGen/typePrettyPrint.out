Value.bar(values: ["hello"])
Rect(location: Point(x: 0, y: 0), size: Size(width: 15, height: 30))
Foo.baz(100)
